### DISCUSSION TITLE
To not show the item groups without permissions

### DISCUSSION THREAD

**Discussion by markkuleinio** [netbox-community/netbox#15293] [Q&A] [answered]
Currently, when the user doesn't have access to for example IPAM>Services, there is an empty Services box in the device view:

![image](https://github.com/netbox-community/netbox/assets/25136274/77691c5f-3b53-4345-a36c-ef0be9d21330)

If the user has access to Services but there aren't any for the device, the item group (maybe I'm using a wrong term, I'm not a Djanginst) is shown like this:

![image](https://github.com/netbox-community/netbox/assets/25136274/8767792c-cb06-4709-95a8-366f5079173c)

Before opening a feature request, how feasible would it be to just omit the whole Services box in the view if the user doesn't have the permissions? Or is there a technical limitation in the background that the (empty) Services box cannot be easily omitted in the view?

I mean, there is no Services item in the main menu either, when the user has no permission for it, so it would be logical to omit the item group as well.

The reason for not having the Services permission is to just keep the GUI a bit more clear for the common user = to remove the unused items in the menus in this particular NetBox setup where Services (or Virtual Device Contexts or Locations or ...) are not used.

FWIW, this is how it looks in netbox.log every time the user looks at a device and the Services box is trying to be populated without permission:

```
2024-02-28 08:24:22,840 django.request WARNING: Forbidden (Permission denied): /ipam/services/
Traceback (most recent call last):
  File "/opt/netbox/venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/netbox/venv/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/netbox/venv/lib/python3.11/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/netbox/netbox/netbox/views/generic/base.py", line 77, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/netbox/netbox/utilities/views.py", line 97, in dispatch
    return self.handle_no_permission()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/netbox/venv/lib/python3.11/site-packages/django/contrib/auth/mixins.py", line 48, in handle_no_permission
    raise PermissionDenied(self.get_permission_denied_message())
django.core.exceptions.PermissionDenied
```

**Comment by abhi1693** [comment, marked as answer]
You can raise this request, it's easy to implement the change in the templates

**Reply by markkuleinio** [reply]
Thanks for the comment, #15294 submitted.
